Sources/TuistKit/Generator/Generator.swift

```diff
@@ -0,0 +1,74 @@
+import Basic
+import Foundation
+import TuistCore
+
+struct GeneratorConfig {
+    
+    static let `default` =  GeneratorConfig()
+    
+    var options: GenerationOptions
+    var directory: GenerationDirectory
+    
+    init(options: GenerationOptions = GenerationOptions(),
+         directory: GenerationDirectory = .manifest) {
+        self.options = options
+        self.directory = directory
+    }
+}
+
+protocol Generating {
+    func generateProject(at path: AbsolutePath, config: GeneratorConfig) throws -> AbsolutePath
+    func generateWorkspace(at path: AbsolutePath, config: GeneratorConfig) throws -> AbsolutePath
+}
+
+extension Generating {
```

Comment:
This particular extension will stay in `TuistKit` - I can possibly split it out to `Extensions` now - wasn't too sure what would be clearer